File: samples/java/src/main/java/co/windly/limbosample/presentation/main/MainModule.java
package co.windly.limbosample.presentation.main;

import dagger.Binds;
import dagger.Module;
import dagger.android.AndroidInjector;
import dagger.multibindings.ClassKey;
import dagger.multibindings.IntoMap;

@Module(subcomponents = { MainComponent.class })
public abstract class MainModule {

  //region Binding

  @Binds
  @IntoMap
  @ClassKey(MainActivity.class)
  abstract AndroidInjector.Factory<?> bindAndroidInjectorFactory(MainComponent.Builder builder);

  //endregion
}

File: samples/kotlin/src/main/java/co/windly/limbosample/presentation/main/MainModule.kt
package co.windly.limbosample.presentation.main

import co.windly.limbosample.presentation.main.MainComponent.Factory
import dagger.Binds
import dagger.Module
import dagger.android.AndroidInjector
import dagger.multibindings.ClassKey
import dagger.multibindings.IntoMap

@Module(subcomponents = [
  MainComponent::class
])
abstract class MainModule {

  //region Binding

  @Binds
  @IntoMap
  @ClassKey(MainActivity::class)
  abstract fun bindAndroidInjectorFactory(factory: Factory): AndroidInjector.Factory<*>

  //endregion
}
